plugins/hls-fourmolu-plugin/src/Ide/Plugin/Fourmolu.hs

```diff
@@ -56,35 +58,46 @@ properties =
 
 provider :: Recorder (WithPriority LogEvent) -> PluginId -> FormattingHandler IdeState
 provider recorder plId ideState typ contents fp fo = withIndefiniteProgress title Cancellable $ do
-    fileOpts <-
-        maybe [] (convertDynFlags . hsc_dflags . hscEnv)
-            <$> liftIO (runAction "Fourmolu" ideState $ use GhcSession fp)
+    dynFlags <- fmap (hsc_dflags . hscEnv) <$> liftIO (runAction "Fourmolu" ideState $ use GhcSession fp)
+    let fileOpts = maybe [] convertDynFlags dynFlags
+        packages =
+            maybe
+                []
+                ( map (T.unpack . T.intercalate "-" . reverse . drop' 2 . reverse . T.splitOn "-" . printOutputable)
+                    . mapMaybe getUnit . packageFlags
+                )
+                dynFlags
+          where
+            getUnit = \case
+                ExposePackage _ (UnitIdArg (RealUnit (Definite (UnitId s)))) _ -> Just s
+                _ -> Nothing
+            drop' n xs = drop (n `min` (Prelude.length xs - 1)) xs
```

Comment:
This is a horrible hack, but I don't have any better ideas. I wouldn't be surprised if there's no clean way to go from GHC `package-id` strings to Hackage library names.